common/tools/eslint-plugin-azure-sdk/src/rules/ts-package-json-sdktype.ts

```diff
@@ -17,41 +17,46 @@ import { getRuleMetaData, stripPath } from "../utils";
 export = {
   meta: getRuleMetaData(
     "ts-package-json-sdktype",
-    "force package.json's sdk-type value to contain be 'client' or 'mgmt'"
+    "force package.json's sdk-type to exist and for its value to be 'client' or 'mgmt'",
+    "code"
   ),
-  create: (context: Rule.RuleContext): Rule.RuleListener =>
-    stripPath(context.getFilename()) === "package.json"
+  create: (context: Rule.RuleContext): Rule.RuleListener => {
+    const verifiers = getVerifiers(context, {
+      outer: "sdk-type"
+    });
+    return stripPath(context.getFilename()) === "package.json"
       ? ({
-          // callback functions
-
-          // check the node corresponding to sdk-type to see if its value contains "client" or "mgmt"
-          "ExpressionStatement > ObjectExpression > Property[key.value='sdk-type']": (
-            node: Property
-          ) => {
-            if (!node) {
-              // Track1 packages don't have this property. Stop checking
-              return;
-            }
-
-            const { value } = node;
-            if (value.type !== "Literal" || typeof value.value !== "string") {
-              context.report({
-                node: node.value,
-                message: "sdk-type is not set to a string"
-              });
-              return;
-            }
-
-            const strValue = stripPath(value.value);
-
-            if (!["client", "mgmt"].includes(strValue)) {
-              context.report({
-                node: node.value,
-                message: "sdk-type is not set to `client` or `mgmt`"
-              });
-              return;
-            }
+        // callback functions
+
+        // check to see if package.json includes 'sdk-type'
+        "ExpressionStatement > ObjectExpression": verifiers.existsInFile,
+
+        // check the node corresponding to sdk-type to see if its value contains "client" or "mgmt"
+        "ExpressionStatement > ObjectExpression > Property[key.value='sdk-type']": (
+          node: Property
+        ): void => {
+          const { value } = node;
+
+          // check for valid type
+          if (value.type !== "Literal" || typeof value.value !== "string") {
+            context.report({
+              node: node.value,
+              message: "sdk-type is not set to a string"
+            });
+            return;
           }
-        } as Rule.RuleListener)
-      : {}
+
+          const strValue = stripPath(value.value);
+
+          if (!["client", "mgmt"].includes(strValue)) {
+            context.report({
+              node: node.value,
+              message: "sdk-type is not set to `client` or `mgmt`"
```

Comment:
Perhaps it could be useful to print the unrecognized value too?
```suggestion
              message: `unrecognized sdk-type value: ${strValue}. Expected `client` or `mgmt` instead.`
```

Please note the error messages in the tests will need to be updated too.